tuning/sft_trainer.py

```diff
@@ -129,9 +129,18 @@ def train(
     )
     
     # load the data by parsing JSON
-    json_dataset = datasets.load_dataset('json', data_files=data_args.data_path)
-    formatted_dataset = json_dataset['train'].map(lambda example : {f"{data_args.dataset_text_field}" : example[f"{data_args.dataset_text_field}"] + tokenizer.eos_token})
-    logger.info(f"Dataset length is {len(formatted_dataset)}")
+    data_files = {"train": data_args.data_path}
+    if data_args.validation_data_path:
+        data_files["validation"] = data_args.validation_data_path
+
+    json_dataset = datasets.load_dataset('json', data_files=data_files)
+    formatted_train_dataset = json_dataset['train'].map(lambda example : {f"{data_args.dataset_text_field}" : example[f"{data_args.dataset_text_field}"] + tokenizer.eos_token})
+    logger.info(f"Dataset length is {len(formatted_train_dataset)}")
+
+    formatted_validation_dataset = None
+    if data_args.validation_data_path:
+        formatted_validation_dataset = json_dataset['validation'].map(lambda example : {f"{data_args.dataset_text_field}" : example[f"{data_args.dataset_text_field}"] + tokenizer.eos_token})
```

Comment:
Can this lambda be assigned to a var and passed to the `map` for both train and eval, just so we don't accidentally change one and not the other later on?

Reply:
Yeah! I was thinking of doing that but was confused on how lamdas work with the `.map()` any pointers would be helpful!

Reply:
You can put the lambda in a variable and pass it to the dataset's `.map`! Here is an example (stubbed the data args and tokenize so that it's runnable if you point it at your own twitter data, but the last two lines are the relevant ones)

```python
import datasets

#Stubbed stuff 
dataset_text_field = "Tweet text"
class DataArgsStub:
    dataset_text_field = "Tweet text"
class TokenizerStub:
    eos_token = "foo"
data_args = DataArgsStub()
tokenizer = TokenizerStub()

data = "twitter_complaints.json"
json_dataset = datasets.load_dataset('json', data_files=data)


format_func = lambda example : {f"{data_args.dataset_text_field}" : example[f"{data_args.dataset_text_field}"] + tokenizer.eos_token}
formatted_train_dataset = json_dataset['train'].map(format_func)
```